Do not force user to create a changes file if version contains a snapshot tag

## Problem

I think it would be nice if the project keeper wouldn't check if the changes file exist or not when the version contains a snapshot suffix. Sometimes I want to increase a version just for testing and I usually add a snapshot there and I don't want to create a changes file for it yet:

[ERROR] E-PK-22 Please change the content of '/exasol-virtual-schema/doc/changes/changes_4.0.0-SNAPHOT.md' by hand!

I had to comment out the keeper plugin to build my test jar.

--- a/doc/changes/changes_0.5.0.md
+++ b/doc/changes/changes_0.5.0.md
@@ -2,6 +2,10 @@
 
 Code name:
 
+## Features / Enhancements
+
+* #52: Changed behavior to not force user to create a changes file if version contains a snapshot tag
+
 ## Bugfixes
 
 * #55: Fixed CVE-2020-8908 by excluding guava dependency

--- a/src/main/java/com/exasol/projectkeeper/validators/ChangelogValidator.java
+++ b/src/main/java/com/exasol/projectkeeper/validators/ChangelogValidator.java
@@ -1,8 +1,6 @@
 package com.exasol.projectkeeper.validators;
 
-import java.io.File;
-import java.io.FileWriter;
-import java.io.IOException;
+import java.io.*;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.time.LocalDateTime;
@@ -32,7 +30,15 @@ public class ChangelogValidator implements Validator {
 
     @Override
     public ChangelogValidator validate(final Consumer<ValidationFinding> findingConsumer) {
-        final String version = getVersionWithoutSnapshotSuffix();
+        final String version = this.mavenProject.getVersion();
+        if (!version.endsWith(SNAPSHOT_SUFFIX)) {
+            validateThatChangesFileExists(findingConsumer, version);
+        }
+        return this;
+    }
+
+    private void validateThatChangesFileExists(final Consumer<ValidationFinding> findingConsumer,
+            final String version) {
         final File changesFile = this.mavenProject.getBasedir().toPath()
                 .resolve(Path.of("doc", "changes", "changes_" + version + ".md")).toFile();
         if (!changesFile.exists()) {
@@ -43,16 +49,6 @@ public class ChangelogValidator implements Validator {
                             getFix(changesFile, changesFileRelativeToProjectRoot)));
         } else {
             checkFileIsDifferentToTemplate(findingConsumer, changesFile);
-        }
-        return this;
-    }
-
-    private String getVersionWithoutSnapshotSuffix() {
-        final String version = this.mavenProject.getVersion();
-        if (version.endsWith(SNAPSHOT_SUFFIX)) {
-            return version.substring(0, version.length() - SNAPSHOT_SUFFIX.length());
-        } else {
-            return version;
         }
     }
 

--- a/src/main/java/com/exasol/projectkeeper/validators/ChangelogValidator.java
+++ b/src/main/java/com/exasol/projectkeeper/validators/ChangelogValidator.java
@@ -31,10 +31,14 @@ public class ChangelogValidator implements Validator {
     @Override
     public ChangelogValidator validate(final Consumer<ValidationFinding> findingConsumer) {
         final String version = this.mavenProject.getVersion();
-        if (!version.endsWith(SNAPSHOT_SUFFIX)) {
+        if (!isSnapshotVersion(version)) {
             validateThatChangesFileExists(findingConsumer, version);
         }
         return this;
+    }
+
+    private boolean isSnapshotVersion(final String version) {
+        return version.endsWith(SNAPSHOT_SUFFIX);
     }
 
     private void validateThatChangesFileExists(final Consumer<ValidationFinding> findingConsumer,

--- a/src/test/java/com/exasol/projectkeeper/validators/ChangelogValidatorTest.java
+++ b/src/test/java/com/exasol/projectkeeper/validators/ChangelogValidatorTest.java
@@ -1,8 +1,7 @@
 package com.exasol.projectkeeper.validators;
 
 import static com.exasol.projectkeeper.FileContentMatcher.hasContent;
-import static com.exasol.projectkeeper.HasValidationFindingWithMessageMatcher.hasValidationFindingWithMessage;
-import static com.exasol.projectkeeper.HasValidationFindingWithMessageMatcher.validationErrorMessages;
+import static com.exasol.projectkeeper.HasValidationFindingWithMessageMatcher.*;
 import static org.hamcrest.CoreMatchers.hasItems;
 import static org.hamcrest.CoreMatchers.startsWith;
 import static org.hamcrest.MatcherAssert.assertThat;
@@ -32,8 +31,7 @@ class ChangelogValidatorTest {
     @Test
     void testValidationForSnapshotVersion(@TempDir final File tempDir) {
         final MavenProject project = createProject(tempDir, TEST_VERSION + "-SNAPSHOT");
-        assertThat(new ChangelogValidator(project),
-                hasValidationFindingWithMessage("E-PK-20 Could not find 'doc/changes/changes_1.2.3.md'."));
+        assertThat(new ChangelogValidator(project), hasNoValidationFindings());
     }
 
     @Test

